src/snowflake/snowpark/mock/_functions.py

```diff
@@ -399,7 +399,12 @@ def convert_date(row):
             else:
                 SnowparkLocalTestingException.raise_from_error(exc)
 
-    res = column.to_frame().apply(convert_date, axis=1)
+    # if there is no data in the column, then apply will not be not executed
+    res = (
```

Comment:
Can we use the approach from: https://github.com/snowflakedb/snowpark-python/pull/1702/files?

Reply:
yes, this is very neat! thanks for the suggestion